packages/geofire-common/src/index.ts

```diff
@@ -64,7 +68,7 @@ export function validateKey(key: string): void {
  *
  * @param location The [latitude, longitude] pair to be verified.
  */
-export function validateLocation(location: number[]): void {
+export function validateLocation(location: geopoint): void {
```

Comment:
If you wanna get fancy you can make these functions actual TS type guards:

```ts
function validateLocation(location: number[]): location is geopoint {
 // ... all the existing validations
 
 return true;
}
```

Reply:
Not sure I follow. How would that strengthen it more than `Geopoint` already does?

Reply:
It just makes the function more "honest" in that what it actually does (based on the content) is check a `number[]` and throw if it's not a valid `geopoint`. Right now we say it takes in a `geopoint` but then it checks things like length which should already be checked.

Doesn't matter either way!

Reply:
Ah, got it. Many of the functions actually have runtime type checks now that are not needed anymore when they are consumed from TypeScript. 

I've been considering how to deal with that, because it's wasteful. But the only solution I came up with so far, would lead to a `.js` and `.ts` deliverable that are significantly different, which felt a bridge too far for now. Especially since the runtime performance impact of the extra checks is negligible when compared to any of the network requests we do.